src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -223,10 +223,10 @@ public SNITCPHandle(string serverName, int port, long timerExpire, object callba
                 }
 
                 _socket.NoDelay = true;
-                _tcpStream = new NetworkStream(_socket, true);
+                _tcpStream = new SNINetworkStream(_socket, true);
```

Comment:
If we're going to have our own versions of NetworkStream and SslStream can we combine some of them into a single class along with SslOverTdsStream?

Reply:
Yes, there's room for improvement.
We can work on modifying design in future PRs, not trivial for this fix to go in for coming release.

Reply:
Sounds sensible.